controller/common/lut.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+from scipy import interpolate
+
+
+class LUT:
+    """
+    Class for performing look-up table interpolation.
+
+    Methods:
+        __init__: Initializes the LUT object with lookup table data and interpolation method.
+        __call__: Calls the interpolation method with the given input.
+    """
+
+    def __init__(self, lookup_table: list, interpolation_method: str = "linear"):
```

Comment:
When you are type hinting, you need to import the type from the `typing` module:

```python
from typing import List

def __init__(self, lookup_table: List, ...):
     ...
```

Secondly, the input is a list of lists. In other words, the type is `List[List[int | float]]`.

Should you consider also allowing for numpy arrays directly in addition to allowing for lists?

Reply:
added typing

`def __init__(
        self,
        lookup_table: Union[List[List[int | float]], np.ndarray],
        interpolation_method: str = "linear",
    ):`

Reply:
also added type checking to init